main.tf

```
@@ -1,22 +1,21 @@
 locals {
-  bucket     = "${var.stack}-redshift"
   elastic_ip = var.publicly_accessible ? aws_eip.default[0].public_ip : null
-  subnet_ids = length(var.subnet_ids) > 0 ? var.subnet_ids : aws_subnet.public[*].id
   vpc_id     = data.aws_subnet.selected.vpc_id
 }
 
 data "aws_subnet" "selected" {
-  id = local.subnet_ids[0]
+  id = var.subnet_ids[0]
 }
 
 resource "aws_eip" "default" {
   count = var.publicly_accessible ? 1 : 0
   vpc   = true
-  tags  = merge(var.tags, map("Name", "${var.stack}-redshift"))
+  tags  = merge(var.tags, map("Name", "${var.name}-redshift"))
 }
 
 resource "aws_security_group" "default" {
-  name        = "${var.stack}-redshift"
+  count       = var.publicly_accessible ? 1 : 0
```

I don't think this is correct, as this group is not used specifically for external access. Instead you could do something like this if you want to make this resource (which I don't really think makes sense) optional:

```suggestion
  count       = var.cidr_blocks != null ? 1 : 0
```

And then give `var.cidr_blocks` a default value of `null` making it optional. But again, I doubt if it should be optional, you always need to assign at least one group so you have access right? Also if you only want internal access.

---

I don't recall the exact reasoning but I think I added this because i only needed it for external access. Will re-visit this

---

the sec group can be omitted but you're right in the sense that it's used for more than public access. Have changed this.